cmd/kubectl-gadget/top/block-io.go

```diff
@@ -30,84 +31,137 @@ import (
 	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/biotop/types"
 )
 
-var blockIONodeStats map[string][]types.Stats
+type BlockIOFlags struct {
+	CommonTopFlags
 
-// flags
-var blockIOSortBy types.SortBy
+	ParsedSortBy types.SortBy
```

Comment:
Make sense in the theory, but I don't see how to do that as `CommonTopFlags.SortBy` is used to store the flag [here](https://github.com/kinvolk/inspektor-gadget/blob/jose/use-parser-top-cat/cmd/kubectl-gadget/top/top.go#L54). Do you have any suggestion?

Reply:
Hmm. You could use `command.Flags().String(...)` in `addCommonTopFlags` and then in `PreRunE` use `cmd.Flags().GetString("sort")` - that helps with the ambiguity but doesn't really make it nicer.

Reply:
Mmmm it would prevent us from defining `SortBy` and also `ParsedSortBy` but it am not sure I like treating some flags in one way (`StringVarP`) and some others in a different way (`String`). IMO, it is not worth it.

Reply:
Agree. Just for the record (and because it might become handy or we change it later on), there would be yet another way: one could use `command.Flags().Var(...)` and then a variable of type `types.SortBy` directly, but would need to implement the `Value` interface for types.SortBy:

```
type Value interface {
	String() string
	Set(string) error
	Type() string
}
```

Reply:
TIL. Thanks for sharing it.